frontend/app/routes/_public+/apply+/$id+/demographics-part1.tsx

```diff
@@ -29,25 +36,107 @@ export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
+
+  if (!parsedDataResult.success) {
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof formSchema>>,
+    });
+  }
+
+  const sessionResponseInit = await applyFlow.saveState({
+    request,
+    params,
+    state: { demographicsPart1: parsedDataResult.data },
+  });
+
+  return redirect(`/apply/${id}/review-information`, sessionResponseInit);
 }
 
 export default function DemographicsPart1() {
-  const { bornTypes, disabilityTypes } = useLoaderData<typeof loader>();
+  const { bornTypes, disabilityTypes, otherEquityCode, equityTypes, state } = useLoaderData<typeof loader>();
   const { i18n, t } = useTranslation(handle.i18nNamespaces);
+  const [otherEquityChecked, setOtherEquityChecked] = useState(state?.otherEquity === otherEquityCode.id);
+  const actionData = useActionData<typeof action>();
+  const errorSummaryId = 'error-summary';
+
+  /**
+   * Gets an error message based on the provided internationalization (i18n) key.
+   *
+   * @param errorI18nKey - The i18n key for the error message.
+   * @returns The corresponding error message, or undefined if no key is provided.
+   */
+  function getErrorMessage(errorI18nKey?: string): string | undefined {
+    if (!errorI18nKey) return undefined;
+
+    /**
+     * The 'as any' is employed to circumvent typechecking, as the type of
+     * 'errorI18nKey' is a string, and the string literal cannot undergo validation.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return t(`demographics-oral-health-questions:part2.error-message.${errorI18nKey}` as any);
+  }
+
+  const errorMessages = {
+    equity: getErrorMessage(actionData?.errors.otherEquity?._errors[0]),
+  };
+
+  const errorSummaryItems = createErrorSummaryItems(errorMessages);
+  useEffect(() => {
+    if (actionData?.formData && hasErrors(actionData.formData)) {
+      scrollAndFocusToErrorSummary(errorSummaryId);
+    }
+  }, [actionData]);
+
+  const nonOtherEquityTypeOptions: InputCheckboxesProps['options'] = equityTypes
+    .filter((equityType) => equityType.id !== otherEquityCode.id)
+    .map((equityType) => ({
+      children: getNameByLanguage(i18n.language, equityType),
+      value: equityType.id,
+      id: equityType.id,
+    }));
+
+  const options: InputCheckboxesProps['options'] = [
+    ...nonOtherEquityTypeOptions,
+    {
+      children: getNameByLanguage(i18n.language, otherEquityCode),
+      value: otherEquityCode.id,
+      defaultChecked: state?.otherEquity === otherEquityCode.id,
+      append: otherEquityChecked && (
+        <div className="mb-4 grid max-w-prose gap-6 md:grid-cols-2 ">
+          <InputField id="otherEquity" type="text" className="w-full" label={t('apply:demographics-oral-health-questions.part1.question3-other-specify')} name="otherEquityFieldName" defaultValue={state?.otherEquity} />
+        </div>
+      ),
+      onClick: () => setOtherEquityChecked(!otherEquityChecked),
```

Comment:
```suggestion
      onClick: () => setOtherEquityChecked((checked) => !checked),
```

See <https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state>